db/migrate/20210326085937_remove_unused_activity_form_states.rb

```diff
@@ -0,0 +1,11 @@
+class RemoveUnusedActivityFormStates < ActiveRecord::Migration[6.0]
+  def up
+    Activity.where(form_state: [nil, :blank, :level, :parent]).destroy_all
```

Comment:
Is this safe to do in production?

Reply:
It would be impossible to know what needs to be added to these activities to make them valid, as none have a `parent_id` yet and most don't even know if they're a fund, programme, project or third-party project.

They're very likely to have been created in error - i.e. clicking "Add activity" and then abandoning the form.

Since, https://github.com/UKGovernmentBEIS/beis-report-official-development-assistance/pull/1004 was merged, there is no way for an end user to create an activity without a `level` or `parent` being set, so this won't affect activities being currently created.

Reply:
I think this is fine, but I'm always a bit iffy about seeing model code in a DB migration. Would this be better as a data migration?

Reply:
If your development database contains any of the above form_states, you'll get weird errors when editing them (because the validations have changed to always expect `form_state` to be set).

Also, wouldn't the next line `change_column_null(:activities, :form_state, false)`  break if any of the existing records contained `form_state == nil` ?

I don't see any harm in doing this here. The only activities that will ever be affected on in obsolete states (i.e. likely most of the ones that show up as `Untitled`).

Reply:
OK, cool. That sounds fine 👍